### PR TITLE
Record dependencies between values from the same lifted_constant pack

### PR DESCRIPTION
**Pull request by chambart**
Definitions in `Lifted_constant.t` have to kept or removed all at once. This adds a data_flow dependency to represent that.

This fixes the problem from https://github.com/ocaml-flambda/flambda-backend/pull/180#issuecomment-915272174

### CODE REVIEW

**Review by Gbury — APPROVED** [review]

**Comment by Gbury** [comment]
Could you also add the repro test before merging this one ?

**Review by mshinwell — APPROVED** [review]

**Comment by mshinwell** [comment]
OK to merge once a test case is added.

**Pull request merged**
